internal/db/model.go

```diff
@@ -0,0 +1,95 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"gorm.io/datatypes"
+)
+
+type Consumer struct {
+	Id     string         `json:"id"`
+	Labels datatypes.JSON `gorm:"column:payload;type:jsonb"`
+}
+
+// had to add a T to prevent clash
+type TResource struct {
+	Id           string         `json:"id"`
+	ConsumerId   string         `json:"consumerId"`
+	GenerationId int64          `json:"generationId"`
+	Object       datatypes.JSON `gorm:"column:object;type:jsonb"`
+	Status       datatypes.JSON `gorm:"column:status;type:jsonb"`
+}
```

Comment:
Is it possible to reuse the existing `Resource` , looks like the `Object` and `Status` field are different?

Reply:
@morvencao i am basically trying to keep it same as the proto structure at start. The proto structure is:
```
.....
	Id           string           `protobuf:"bytes,1,opt,name=id,proto3" json:"id,omitempty"`
	ConsumerId   string           `protobuf:"bytes,2,opt,name=consumerId,proto3" json:"consumerId,omitempty"`
	GenerationId int64            `protobuf:"varint,3,opt,name=generationId,proto3" json:"generationId,omitempty"`
	Object       *structpb.Struct `protobuf:"bytes,4,opt,name=object,proto3" json:"object,omitempty"`
	Status       *structpb.Struct `protobuf:"bytes,5,opt,name=status,proto3" json:"status,omitempty"`
}
```

Does that make sense?

Reply:
thanks, it makes sense.